opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Export DataSourcePluginRequestContext at top level for plugins to use

### Description

This change export DataSourcePluginRequestContext at top level for external plugins to consume

### Issues Resolved

Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6045

## Screenshot


https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/70f912c3-e6bd-4120-ae4a-8bc228871b00



## Testing the changes
The following were performed in the recording:
1. add dataSource with type exported from data source plugin into external plugin
2. go to the handler, should not see compiler error, and then go to the reference will route to the current type

### Check List
